### ISSUE TITLE
insufficient resources

### ISSUE THREAD

**Issue by jedmangubat** [stefan2200/TWB#49]
my bot does not build anything. it says not building anything because insufficient resources even though I have enough resources.

**Comment by suchencjusz** [comment]
After my quick inspection it seems to me that this could be the cause of the problem

https://github.com/stefan2200/TWB/blob/046aa22cb2d862f6e921b950263bd1d9c56f6eb9/core/extractors.py#L25C18-L31

is not returning build links anymore, aren't even in the part where extractor try to finds them 
I'm not sure if the bot works the way I think it does

**Comment by jedmangubat** [comment]
so are there any fix?

**Comment by suchencjusz** [comment]
I fixed it here https://github.com/stefan2200/TWB/pull/50#issue-1851213407

**Comment by jedmangubat** [comment]
it's working now. thank you so much!

**Comment by jedmangubat** [comment]
there is another issue. it only builds 1 but for the 2nd queue, it also says not building anything because insufficient resources even though I have enough resources.

**Comment by stefan2200** [comment]
Hey @suchencjusz ,

Thanks for looking into this!
I think this is caused by a UI change that blocks the building data from being loaded correctly.

I'll approve your merge request since it appears to work on all international versions!

Thanks again and kind regards,
Stefan

**Comment by suchencjusz** [comment]
Hi
Thanks for merging, 

I discovered that a similar problem occurs with premuim marketplace.
I'll look into it at my leisure

**Comment by jedmangubat** [comment]
it's back to not building anything. fyi. thank you for dealing with this.

**Comment by suchencjusz** [comment]
> it's back to not building anything. fyi. thank you for dealing with this.

In my case, it works well.
![image](https://github.com/stefan2200/TWB/assets/34921955/2de66c70-6f64-4d81-80f8-410ea7e01865)

@jedmangubat  What world do you play on? Send me any diagnostic data

**Comment by jedmangubat** [comment]
it only builds 1 then says insufficient on 2nd building.

![image](https://github.com/stefan2200/TWB/assets/133950725/de0db83f-ac44-4688-b170-4b7b68b42a60)

i edited buildingmanager.py as per the edited file in here.

![image](https://github.com/stefan2200/TWB/assets/133950725/972f0b7a-ec89-4bf6-9849-4f0f30201dc2)

i don't know how to do this one.

![image](https://github.com/stefan2200/TWB/assets/133950725/fcd0c875-e8bc-4f1b-89f4-8c0043c5522a)

please instruct me if i'm doing something wrong.

**Comment by suchencjusz** [comment]
Ok I already know what the issue is, for some reason, the building upgrade is added to the list only once, I honestly have no idea what the reason is but anyway, the next time it cycles the building will add to the queue.

**Comment by jedmangubat** [comment]
yes it does. but that's just because i haven't activated premium account yet. it will wait for cycles to add 5 building queues. well, i know i'll just wait for you to fix this as i know very little to none about coding especially python. thanks.

**Comment by stefan2200** [comment]
The issue was that the game state changed and was not executed again after the first building action was completed. Re-running it fixed adding more items to the queue.
